src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java

```diff
@@ -313,6 +315,24 @@ public Object convert(Class type, Object value) {
 
     }
 
+    /** A converter that converts a comma-separated string into an array of patterns. */
+    private static final class PatternArrayConverter implements Converter {
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public Object convert(Class type, Object value) {
+            final String plainString = value.toString();
+            Pattern[] patterns = new Pattern[0];
+            if (!CommonUtil.isBlank(plainString)) {
+                final String[] patternStrings = plainString.split(COMMA_SEPARATOR);
```

Comment:
Sorry I missed this before.

https://github.com/checkstyle/checkstyle/blob/8dc065a6cbe028141ba0f863d14d95ecae4675ef/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java#L393
https://github.com/checkstyle/checkstyle/blob/8dc065a6cbe028141ba0f863d14d95ecae4675ef/src/main/java/com/puppycrawl/tools/checkstyle/AbstractAutomaticBean.java#L398
This is similar to `RelaxedStringArrayConverter` and it trims the input during the split. You are not.

You can see in `testBeanConverterPatternArray` that the space remains in the regular expression. This violates your example you added in `property_types.xml` as it shows the space removed.

Originally we were using `RelaxedStringArrayConverter` for these type of instances, so we should continue the trimming it does. We need to implement trim in this method. I would expect spaces to be trimmed.

Reply:
> Sorry I missed this before.
Totally no worries. This is a great catch. I was about to ask about this, but as time went by, I forgot...

Added trim. I have also updated the new converter to have the same structure with `RelaxedStringArrayConverter`, so we can reduce one import to `Arrays`.